tmk_core/protocol/chibios/usb_main.c

```diff
@@ -930,9 +930,20 @@ void send_consumer(uint16_t data) {
 #ifdef CONSOLE_ENABLE
 
 int8_t sendchar(uint8_t c) {
-    // The previous implmentation had timeouts, but I think it's better to just slow down
-    // and make sure that everything is transferred, rather than dropping stuff
-    return chnWrite(&drivers.console_driver.driver, &c, 1);
+    static bool timeouted = false;
+    if (!timeouted) {
+        size_t result = chnWriteTimeout(&drivers.console_driver.driver, &c, 1, TIME_MS2I(5));
```

Comment:
When thinking this through again, it occurred to me that this line as written would incur 5ms of additional latency in key processing when `hid_listen` is not running, no?

If I’m correct in this assumption, then that sounds like something to avoid.

Reply:
This latency occurs only once, the first time the write doesn't succeed. It times out, sot chnWriteTimeout returns zero, and we set timeouted = true. And all further characters will be written with TIME_IMMEDIATE, which is the same as a zero timeout (and if hid_listen is not started, all of these will fail too, and we never go back to the timeouted=false state). When we start hid_listen after a while, suddenly the TIME_IMMEDIATE writes start succeeding, so only then we go back to the timeouted=false state. This is exactly how it works on AVR.

Reply:
Aha, so the conditional lazily adjusts the code path. Thanks for the explanation. Can you add that as a comment, too, please?

Reply:
I added a C comment describing the logic. Explained a bit differently then how I explained in the above Github comment.

Reply:
Resolved, comments added.